bin/schemas/story.json

```diff
@@ -0,0 +1,839 @@
+          "$ref": "#/$defs/element"
+        }
+      ],
+      "required": ["resource"],
+      "properties": {
+        "resource": {
+          "type": "object",
+          "title": "The resource Schema",
+          "required": [
+            "type",
+            "mimeType",
+            "src",
+            "width",
+            "height",
+            "id",
+            "alt",
+            "sizes"
+          ],
+          "properties": {
+            "type": {
+              "type": "string",
+              "title": "Resource type",
+              "enum": ["image", "video", "gif"]
+            },
+            "mimeType": {
+              "type": "string",
+              "title": "Mime type"
+            },
+            "src": {
+              "type": "string",
+              "title": "Media source"
+            },
+            "width": {
+              "type": "integer",
+              "title": "Media width",
+              "minimum": 1
+            },
+            "height": {
+              "type": "integer",
+              "title": "Media height",
+              "minimum": 1
+            },
+            "id": {
+              "type": "integer",
+              "title": "Resource ID"
+            },
+            "alt": {
+              "type": "string",
+              "title": "Alt text"
+            },
+            "sizes": {
+              "type": "array",
+              "title": "The sizes Schema",
+              "items": {},
+              "$comment": "TODO: Complete schema"
+            },
+            "isOptimized": {
+              "type": "boolean",
+              "title": "Whether the resource is considered optimized"
+            },
+            "baseColor": {
+              "type": "string",
+              "title": "Average/Base color",
+              "examples": ["#99684c", "#daa992"]
+            },
+            "isExternal": {
+              "type": "boolean",
+              "title": "Whether it is an external resource"
+            },
+            "creationDate": {
+              "type": "string",
+              "default": "",
+              "title": "Date the media file was uploaded",
+              "$comment": "TODO: Remove? Use date-time format and ajv-formats?"
+            },
+            "posterId": {
+              "type": "integer",
+              "default": 0,
+              "title": "ID of the resource's poster"
+            }
+          }
+        },
+        "overlay": {
+          "type": "object",
+          "$ref": "#/$defs/linearPattern"
+        }
+      }
+    },
+    "shapeElement": {
+      "type": "object",
+      "title": "Shape element",
+      "allOf": [
+        {
+          "$ref": "#/$defs/element"
+        }
+      ]
+    },
+    "pattern": {
+      "type": "object",
+      "oneOf": [
+        { "$ref": "#/$defs/solidPattern" },
+        { "$ref": "#/$defs/linearPattern" }
+      ]
+    },
+    "solidPattern": {
+      "type": "object",
+      "title": "Solid pattern",
+      "required": ["color"],
```

Comment:
<!-- propose improvements -->
**💬 suggestion:**

"type" is required here, too, AFAIK.

Reply:
In the templates most text elements have bg color defined like this:

`"backgroundColor": { "color": { "r": 196, "g": 196, "b": 196 } },`

Without `type`.

Is that a different type of pattern? Otherwise I guess it's optional.

Reply:
Ah, you're right, looks like 'solid' is the default if not set.

Reply:
I should list that property though for `solidPattern`, so good call